api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -201,32 +192,40 @@ public void updateProfileForUser(DbUser user, Profile updatedProfile, Profile pr
     user.setDemographicSurvey(dbDemographicSurvey);
     user.setLastModifiedTime(now);
 
-    updateInstitutionalAffiliations(updatedProfile, user);
     boolean requireInstitutionalVerification =
         workbenchConfigProvider.get().featureFlags.requireInstitutionalVerification;
     if (requireInstitutionalVerification) {
       validateInstitutionalAffiliation(updatedProfile);
+    } else {
+      updateInstitutionalAffiliations(updatedProfile, user);
     }
 
-    userService.updateUserWithConflictHandling(user);
+    user = userService.updateUserWithConflictHandling(user);
+
     if (requireInstitutionalVerification) {
-      DbVerifiedInstitutionalAffiliation updatedDbVerifiedAffiliation =
-          verifiedInstitutionalAffiliationMapper.modelToDbWithoutUser(
-              updatedProfile.getVerifiedInstitutionalAffiliation(), institutionService);
-      updatedDbVerifiedAffiliation.setUser(user);
-      Optional<DbVerifiedInstitutionalAffiliation> dbVerifiedAffiliation =
-          verifiedInstitutionalAffiliationDao.findFirstByUser(user);
-      dbVerifiedAffiliation.ifPresent(
-          verifiedInstitutionalAffiliation ->
-              updatedDbVerifiedAffiliation.setVerifiedInstitutionalAffiliationId(
-                  verifiedInstitutionalAffiliation.getVerifiedInstitutionalAffiliationId()));
-      this.verifiedInstitutionalAffiliationDao.save(updatedDbVerifiedAffiliation);
+      saveVerifiedInstitutionalAffiliation(user, updatedProfile);
     }
 
     final Profile appliedUpdatedProfile = getProfile(user);
     profileAuditor.fireUpdateAction(previousProfile, appliedUpdatedProfile);
   }
 
+  public void saveVerifiedInstitutionalAffiliation(DbUser user, Profile updatedProfile) {
```

Comment:
Please split this per the [law of demeter](https://en.wikipedia.org/wiki/Law_of_Demeter). I.e. I'd expect this method to take in only a DbUser and an Affiliation ID. Since that's all it needs according to the name, the rest is confusing.

Also "updated profile" is a description that only makes sense within a function. Updated how?